src/AppInstallerRepositoryCore/Microsoft/PredefinedInstalledSourceFactory.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "Microsoft/PredefinedInstalledSourceFactory.h"
+#include "Microsoft/SQLiteIndex.h"
+#include "Microsoft/SQLiteIndexSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Microsoft
+{
+    namespace
+    {
+        // Populates the index with the ARP entries from the given root.
+        void PopulateIndexFromARP(SQLiteIndex& index, HKEY rootKey)
+        {
+            UNREFERENCED_PARAMETER(index);
+            UNREFERENCED_PARAMETER(rootKey);
+        }
+
+        // Populates the index with the entries from MSIX.
+        void PopulateIndexFromMSIX(SQLiteIndex& index)
+        {
+            using namespace winrt::Windows::ApplicationModel;
+            using namespace winrt::Windows::Management::Deployment;
+
+            PackageManager packageManager;
+            auto packages = packageManager.FindPackagesForUserWithPackageTypes({}, PackageTypes::Main);
+
+            // Reuse the same manifest object, as we will be setting the same values every time.
+            Manifest::Manifest manifest;
+            // Add one installer for storing the package family name.
+            manifest.Installers.emplace_back();
+            // Every package will have the same tags currently.
+            manifest.Tags = { "msix" };

```

Comment:
>manifest.Tags = { "msix" }; [](start = 12, length = 27)

What's this tag used for?

Reply:
My expectation was that it would allow `list msix`, but there could be a better way to allow that in the future.

---
In reply to: [507498234](https://github.com/microsoft/winget-cli/pull/598#discussion_r507498234) [](ancestors = 507498234)